lib/https.js

```diff
@@ -182,8 +182,42 @@ exports.request = function(options, cb) {
   return http.request(options, cb);
 };
 
+exports.requestAsync = function(options) {
+  var resolveRequest = null;
+  const getResponse = new Promise((resolve) => {
+    resolveRequest = resolve;
+  });
+  const request = exports.request(options, (res) => {
+    resolveRequest(res);
+  });
+  return {
+    request,
+    response: getResponse
+  };
+};
+
 exports.get = function(options, cb) {
-  var req = exports.request(options, cb);
+  const req = exports.request(options, cb);
   req.end();
   return req;
 };
+
+exports.getAsync = function(options) {
+  var resolveRequest = null;
+  var rejectRequest = null;
+  const getResponse = new Promise((resolve, reject) => {
+    resolveRequest = resolve;
+    rejectRequest = reject;
+  });
+  const request = exports.request(options, (res) => {
+    resolveRequest(res);
+  });
+  request.on('error', rejectRequest);
+  getResponse.then(() => {
+    request.removeListener('error', rejectRequest);
```

Comment:
As @inikulin pointed in a different file, `onerror` should be removed in any case, otherwise we introduce memory leak.